clippy_lints/src/methods/mod.rs

```diff
@@ -1796,6 +1797,30 @@ declare_clippy_lint! {
     "replace `.splitn(2, pat)` with `.split_once(pat)`"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Check for `&mut iter` followed by a method call.
+    ///
+    /// ### Why is this bad?
+    /// This requires using parenthesis to signify precedence.
+    ///
+    /// ### Example
+    /// ```rust
+    /// let mut iter = ['a', 'b', '.', 'd'].iter();
+    /// let before_dot = (&mut iter).take_while(|&&c| c != '.').collect::<Vec<_>>();
+    /// let after_dot = iter.collect::<Vec<_>>();
+    /// ```
+    /// Use instead:
+    /// ```rust
+    /// let mut iter = ['a', 'b', '.', 'd'].iter();
+    /// let before_dot = iter.by_ref().take_while(|&&c| c != '.').collect::<Vec<_>>();
+    /// let after_dot = iter.collect::<Vec<_>>();
+    /// ```
+    pub REF_MUT_ITER_METHOD_CHAIN,
```

Comment:
The PR #7813 was just merged, which added a new `clippy::version` attribute to all lints. Just a heads-up that the attribute needs to added for a green CI after the next rebase :upside_down_face: 
```suggestion
    #[clippy::version = "1.58.0"]
    pub REF_MUT_ITER_METHOD_CHAIN,
```